lib/AST/Decl.cpp

```diff
@@ -1357,6 +1357,46 @@ bool ValueDecl::needsCapture() const {
   return true;
 }
 
+bool ValueDecl::isTestCandidate() const {
+  if (!this->hasName())
+    return false;
+
+  // A 'test candidate' is:
+  // 1. An instance method...
+  auto FD = dyn_cast<FuncDecl>(this);
+  if (!FD)
+    return false;
+  if (FD->isStatic())
+    return false;
+  if (!this->getDeclContext()->isTypeContext())
+    return false;
+  Type Ty = this->getDeclContext()->getDeclaredTypeOfContext();
+  if (!Ty)
+    return false;
+  auto NTD = Ty->getAnyNominal();
+  if (!NTD)
+    return false;
+  if (!isa<ClassDecl>(NTD))
+    return false;
+
+  // 2. ...that returns void...
+  Type RetTy = FD->getResultType();
+  if (RetTy && !RetTy->isVoid())
+    return false;
+
+  // 3. ...takes no parameters...
+  if (FD->getParameterLists().size() != 2)
+    return false;
+  if (FD->getParameterList(1)->size() != 0)
+    return false;
+
+  // 4. ...and starts with "test".
+  if (FD->getName().str().startswith("test"))
```

Comment:
I would agree with you, but the behavior you describe doesn't match that of Apple XCTest reflection. The Apple XCTest method for finding test methods simply looks for a prefix (and has done so [since the SenTestingKit era](https://github.com/modocache/OCUnit/blob/f609940845d48c23b4629ae990c3395cd0540364/OCUnit/SourceCode/SenTestingKit/NSInvocation_SenTesting.m#L69-L72)). Here's a screenshot of test detection from Xcode 7.3:

![screen shot 2016-05-06 at 11 17 11 am](https://cloud.githubusercontent.com/assets/552921/15082363/2add2b3c-137c-11e6-8b3e-34608244dea6.png)